app/controllers/OneOffContributions.scala

```diff
@@ -38,21 +40,31 @@ class OneOffContributions(
     )
   }
 
-  def displayForm(paypal: Option[Boolean]): Action[AnyContent] = CachedAction() { implicit request =>
-    Ok(
-      oneOffContributions(
-        title = "Support the Guardian | One-off Contribution",
-        id = "oneoff-contributions-page",
-        js = "oneoffContributionsPage.js",
-        payPalButton = paypal.getOrElse(true),
-        defaultStripeConfig = stripeConfigProvider.get(false),
-        uatStripeConfig = stripeConfigProvider.get(true),
-        contributionsStripeEndpoint = contributionsStripeEndpoint,
-        contributionsPayPalEndpoint = contributionsPayPalEndpoint
-      )
+  def getOneOffContributions(idUser: Option[IdUser], paypal: Option[Boolean])(implicit request: RequestHeader): Html = {
+    oneOffContributions(
+      title = "Support the Guardian | One-off Contribution",
+      id = "oneoff-contributions-page",
+      js = "oneoffContributionsPage.js",
+      payPalButton = paypal.getOrElse(true),
+      defaultStripeConfig = stripeConfigProvider.get(false),
+      uatStripeConfig = stripeConfigProvider.get(true),
+      contributionsStripeEndpoint = contributionsStripeEndpoint,
+      contributionsPayPalEndpoint = contributionsPayPalEndpoint,
+      idUser = idUser
     )
   }
 
+  def displayForm(paypal: Option[Boolean]): Action[AnyContent] = MaybeAuthenticatedAction.async { implicit request =>
+    request.user.fold(
+      Future.successful(Ok(getOneOffContributions(None, paypal)))
+    )(u => {
```

Comment:
"Guardian Style" would be to write this as `({ u =>`, and probably a more meaningful name like `minimalUser`